lib/ansible/galaxy/collection.py

```diff
@@ -0,0 +1,923 @@
+# Copyright: (c) 2019, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import fnmatch
+import json
+import os
+import shutil
+import tarfile
+import tempfile
+import time
+import uuid
+import yaml
+
+from contextlib import contextmanager
+from distutils.version import LooseVersion
+from hashlib import sha256
+from io import BytesIO
+from yaml.error import YAMLError
+
+try:
+    from urllib.parse import urlparse  # Python 3
+except ImportError:
+    from urlparse import urlparse  # Python 2
+
+import ansible.constants as C
+import ansible.module_utils.six.moves.urllib.error as urllib_error
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_bytes, to_native, to_text
+from ansible.utils.display import Display
+from ansible.utils.hashing import secure_hash, secure_hash_s
+
+from ansible.module_utils.urls import open_url
+
+
+display = Display()
+
+MANIFEST_FORMAT = 1
+
+
+class CollectionRequirement(object):
```

Comment:
1) No need to inherit from object explicitly due to `__metaclass__ = type` set above
2) It's nice to properly support all stringifications
```suggestion
@six.python_2_unicode_compatible
class CollectionRequirement:
```

Reply:
I've added the decorator but personally prefer being explicit about the base and will keep `(object)`.